docs/structure-of-a-contract.rst

```diff
@@ -31,6 +31,25 @@ See the :ref:`types` section for valid state variable types and
 :ref:`visibility-and-getters` for possible choices for
 visibility.
 
+You can initialize a state variable inline for all types (even for structs). But for arrays you must declare them as static memory arrays. For example::
```

Comment:
`you must declare them as static memory arrays` - those arrays cannot be memory arrays, they have to be storage arrays, and I would just say that initializing dynamically-sized arrays and mappings is not possible inline and has to be done in the constructor instead.

Reply:
@chriseth To check, when you say 

> and I would just say that initializing dynamically-sized arrays and mappings is not possible inline and has to be done in the constructor instead.

That's just for state variables?

Reply:
Yes, just for state variables.